code/obj/item/cigarette.dm

```diff
@@ -996,6 +996,9 @@
 				src.light()
 
 	ex_act(severity)
+		..()
+		if (!istype(src))
```

Comment:
What's this supposed to do? `src` should always be of its given type by definition, if you're trying to check for it being deleted then do `QDELETED(src)` instead